### PR TITLE
Expand login handler responsibilities and capabilities

### PR DESCRIPTION
**Pull request by A248**
This will enable us better flexibility with regards to using the data loader in different environments, including on the proxy and in testing/development

### CODE REVIEW

File: core-implementation/src/main/java/gg/solarmc/loader/impl/Icarus.java [hunk 1]
```diff
@@ -19,25 +19,26 @@
...
-	private final LoginHandler loginHandler;
 	private final TransactionSource transactionSource;
 	private final DataManagementCenter dataManagement;
+	private final Set<DataGroup<?, ?, ?>> groups;
 	private final DataCenterLifecycle lifecycle;
 
-	Icarus(LoginHandler loginHandler, TransactionSource transactionSource,
-		   DataManagementCenter dataManagement, DataCenterLifecycle lifecycle) {
-		this.loginHandler = loginHandler;
+	Icarus(TransactionSource transactionSource, DataManagementCenter dataManagement,
```
Comment (auriium) [inline core-implementation/src/main/java/gg/solarmc/loader/impl/Icarus.java:34]: Why are you moving datagroups into the icarus?

File: core-implementation/src/main/java/gg/solarmc/loader/impl/login/NameAddressHistoryUpdate.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    class NoOpImpl implements NameAddressHistoryUpdate {
+
+        @Override
+        public void update(Transaction transaction, UserDetails userDetails) {
+
+        }
+    }
+
+    class FunctioningImpl implements NameAddressHistoryUpdate {
```
Comment (auriium) [inline core-implementation/src/main/java/gg/solarmc/loader/impl/login/NameAddressHistoryUpdate.java:41]: Does dataloader depend on libertybans now? I know ariel said that we're gonna be using it, and it's a good thing this is interface-based

**Review by auriium — APPROVED** [review]
Everything looks good - Merging this should allow us to finally move away from the core of things and begin work on the frontend

File: core-implementation/src/main/java/gg/solarmc/loader/impl/Icarus.java (hunk 1 above)
Comment (A248) [reply core-implementation/src/main/java/gg/solarmc/loader/impl/Icarus.java:34]: The LoginHandler.Builder needs them. See line 56

File: core-implementation/src/main/java/gg/solarmc/loader/impl/login/NameAddressHistoryUpdate.java (hunk 2 above)
Comment (A248) [reply core-implementation/src/main/java/gg/solarmc/loader/impl/login/NameAddressHistoryUpdate.java:41]: The issue is, we don't want to duplicate name and address history storage. If libertybans were specifically made for solarmc, then the name and address history tables would be in data loader, and libertybans would therefore depend on the data loader tables.

However, using a public plugin is somewhat constraining, and as a result the table dependency has to be the other way around. Of course neither depends on each other's Java API.

**Pull request merged**